ontology_processing/graph_creation/make_graph_class.py

```diff
@@ -0,0 +1,652 @@
+        subgraph_downstream = custom_bfs(
+            self.B_annotated, "increase in greenhouse effect", edge_type="causes_or_promotes"
+        ).copy()
+
+        total_adaptation_nodes = []
+        for effectNode in subgraph_downstream_adaptations.nodes():
+            intermediate_nodes = nx.all_simple_paths(
+                self.B, "increase in greenhouse effect", effectNode
+            )
+            # collapse nested lists and remove duplicates
+            intermediate_nodes = [
+                item for sublist in intermediate_nodes for item in sublist
+            ]
+            intermediate_nodes = list(
+                dict.fromkeys(intermediate_nodes)
+            )  # gets unique nodes
+            node_adaptation_solutions = list()
+            for intermediateNode in intermediate_nodes:
+                node_neighbors = self.G.neighbors(intermediateNode)
+                for neighbor in node_neighbors:
+                    if (
+                        self.G[intermediateNode][neighbor]["type"]
+                        == "is_inhibited_or_prevented_or_blocked_or_slowed_by"
+                    ):  # bad to hard code in 'is_inhibited_or_prevented_or_blocked_or_slowed_by'
+                        node_adaptation_solutions.append(neighbor)
+            # add the adaptation solutions to the networkx object for the node
+            # be sure that solutions don't show up as effectNodes! and that they aren't solutions to themself! the code needs to be changed to avoid this.
+            # ^solutions shouldn't be added as solutions to themself!
+            node_adaptation_solutions = list(
+                OrderedDict.fromkeys(node_adaptation_solutions)
+            )  # gets unique nodes
+
+            # need to add a check here that doesn't add to effectNode attributes the effectNode as an adaptation solution (solution nodes should have themself as an adaptation solution!)
+            nx.set_node_attributes(
+                self.G, {effectNode: node_adaptation_solutions}, "adaptation solutions"
+            )
+
+            # add solution sources field to all adaptation solution nodes
+            for solution in node_adaptation_solutions:
+                sources = solution_sources(self.G.nodes[solution])
+                nx.set_node_attributes(
+                    self.G,
+                    {solution: sources},
+                    "solution sources",
+                )
+            total_adaptation_nodes.extend(node_adaptation_solutions)
+
+        subgraph_adaptations = self.B_annotated.subgraph(total_adaptation_nodes).copy()
+
+        subgraph_upstream_mitigations = union_subgraph(
+            [self.subgraph_upstream, self.subgraph_mitigation], base_graph=self.B_annotated
+        ).copy()
+
+        # Computes an array of elements + nodes to input into cytoscape.js for each personal value
+        # Computes the subgraphs achieved by BFS through each of those personal values
+
+        personal_values = [
```

Comment:
Henry wrote this line I think. I think his name for this variable is poorly named? Is this actually nodes that contain a personal value associated with it?

Reply:
I could be wrong, but I think this is okay. He's getting a list of personal value labels (i.e. "security") and then creating a dictionary where those labels are the keys. Let me know if you're seeing something I'm not.